mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -311,8 +350,28 @@ class LogisticRegression @Since("1.2.0") (
 
     val histogram = labelSummarizer.histogram
     val numInvalid = labelSummarizer.countInvalid
-    val numClasses = histogram.length
     val numFeatures = summarizer.mean.size
+    val numFeaturesPlusIntercept = if (getFitIntercept) numFeatures + 1 else numFeatures
+
+    val numClasses = MetadataUtils.getNumClasses(dataset.schema($(labelCol))) match {
+      case Some(n: Int) =>
+        require(n >= histogram.length, s"Specified number of classes $n was " +
+          s"less than the number of unique labels ${histogram.length}.")
+        n
+      case None => histogram.length
+    }
+
+    val isBinaryClassification = numClasses == 1 || numClasses == 2
+    val isMultinomial = $(family) match {
+      case "binomial" =>
+        require(isBinaryClassification, s"Binomial family only supports 1 or 2 " +
+        s"outcome classes but found $numClasses.")
+        false
+      case "multinomial" => true
+      case "auto" => !isBinaryClassification
+      case other => throw new IllegalArgumentException(s"Unsupported family: $other")
+    }
```

Comment:
`isBinaryClassification` is not the same as `isBinomial`, and it's intended that it could be true along with `isMultinomial`. I can switch it up to not use the `isBinaryClassification` variable, but `isBinomial` is never used so I don't see a need to define it.